src/Field_Properties.hxx

```diff
@@ -42,5 +42,31 @@ public:
               attributes(Attributes),
               values(v),
               links(Links) {}
+
+    inline const std::string &get_description() const { return description; }
+
+    inline const std::map<std::string, std::string> &get_attributes() const {
+        return attributes;
+    }
+    inline std::map<std::string, std::string> &get_attributes() { return attributes; }
+
+    inline const Values &get_values() const { return values; }
+    inline Values &get_values() { return values; }
+
+    inline const std::vector<std::pair<std::string, std::string> > &get_links() const {
+        return links;
+    }
+    inline std::vector<std::pair<std::string, std::string> > &get_links() {
+        return links;
+    }
+
+    inline void set_description(const std::string &desc) { description.assign(desc); }
+    inline void set_attributes(const std::map<std::string, std::string> &attrs) {
+        attributes = attrs;
+    }
+    inline void set_values(const Values &vals) { values = vals; }
+    inline void set_links(const std::vector<std::pair<std::string, std::string> > &ls) {
+        links = ls;
+    }
```

Comment:
I think the same `constexpr` comment applies in the other files.

Reply:
Thanks, will make the constexpr changes.

The nice thing about having public accessors and private members is that you can change the implementation and replace the private members with something altogether different and your clients won't even notice.

Reply:
I tend to agree.

Reply:
The compiler didn't like the way I tried to use constexpr.  I don't understand why it *should* work for getters and setters---isn't the idea that constexpr functions can be evaluated at compile time?

Reply:
I'm guessing this is because there is no constexpr constructor. But it isn't possible to provide one, because the member types don't have them. `const` it is (sorry for the detour).